subgraphs/aave-v2/src/mappings/utilFunctions.ts

```diff
@@ -0,0 +1,605 @@
+    if (!market) return;
+    market.outputTokenSupply = tryTokenSupply.value;
+    market.save();
+    // This funcion only gets called from mint/burn events which only happen following deposits and withdraws which update the market snapshot
+    // Calling the getMarketDailySnapshot function seems to cause some sort of overflow error, so only the outputTokenSupply field needs to be set
+    const snapId = getMarketDailyId(event, market);
+    const snap = MarketDailySnapshot.load(snapId);
+    if (!snap) return;
+    snap.outputTokenSupply = market.outputTokenSupply;
+    snap.save();
+  } else {
+    log.info('OUTPUT TOKEN SUPPLY CALL REVERTED FOR TOKEN ' + outputTokenAddr.toHexString(), []);
+  }
+}
+
+export function initToken (assetAddr: Address): Token {
+  // The token id is set as the asset address
+  let asset = Token.load(assetAddr.toHex());
+  if (asset === null) {
+    // Create a new Token implementation
+    asset = new Token(assetAddr.toHex());
+    // Instantiate the Token with the IERC20 interface in order to access contract read methods
+    const tokenInstance = IERC20.bind(assetAddr);
+    const tryName = tokenInstance.try_name();
+    if (!tryName.reverted) {
+      asset.name = tryName.value;
+    }
+    const trySymbol = tokenInstance.try_symbol();
+    if (!trySymbol.reverted) {
+      asset.symbol = trySymbol.value;
+    }
+    const tryDecimals = tokenInstance.try_decimals();
+    if (!tryDecimals.reverted) {
+      asset.decimals = tryDecimals.value;
+    }
+    asset.save();
+  }
+  return asset as Token;
+}
+
+export function loadRewardToken (assetAddr: Address, market: Market, type: string): RewardToken {
+  // See initToken() function above
+  let asset = RewardToken.load(type + '-' + assetAddr.toHex());
+  if (asset === null) {
+    asset = new RewardToken(type + '-' + assetAddr.toHex());
+    const tokenInstance = IERC20.bind(assetAddr);
+    const tryName = tokenInstance.try_name();
+    if (!tryName.reverted) {
+      asset.name = tryName.value;
+    }
+    const trySymbol = tokenInstance.try_symbol();
+    if (!trySymbol.reverted) {
+      asset.symbol = trySymbol.value;
+    }
+    const tryDecimals = tokenInstance.try_decimals();
+    if (!tryDecimals.reverted) {
+      asset.decimals = tryDecimals.value;
+    }
+    asset.type = type;
+    asset.save();
+  }
+
+  initToken(assetAddr);
+  return asset as RewardToken;
+}
+
+export function fetchProtocolEntity (protocolId: string): LendingProtocol {
+  // Load or create the Lending Protocol entity implementation
+  let lendingProtocol = LendingProtocol.load(protocolId);
+  log.info('proto id: ' + protocolId, []);
+  if (!lendingProtocol) {
+    lendingProtocol = new LendingProtocol(protocolId);
+    lendingProtocol.totalUniqueUsers = 0;
+    lendingProtocol.subgraphVersion = '1.0.0';
+    lendingProtocol.schemaVersion = '1.1.0';
+    lendingProtocol.name = 'Aave-v2';
+    lendingProtocol.slug = 'aave-v2';
+    lendingProtocol.network = 'ETHEREUM';
```

Comment:
Use constants

Reply:
Added values to constants.ts file under 'Protocol Specific' section, set 'lendingProtocol' entity properties to these constants.